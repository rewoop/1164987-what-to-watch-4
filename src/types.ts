export interface Film {
  id: number;
  filmTitle: string;
  filmVideo: string;
  filmGenre: string;
  releaseDate: number;
  backgroundPoster: string;
  filmPoster: string;
  ratingScore: number;
  ratingLevel: number;
  ratingCount: string;
  filmDescription: string;
  filmDirector: string;
  filmStarring: string[];
  filmRunTime: string;
  isFavoriteFilm: boolean;
}

export type Films = Film[];

export interface FilmComment {
  id: number;
  user: {
    id: number;
    name: string;
  };
  comment: string;
  date: string;
  rating: number;
}

export type FilmComments = FilmComment[];

export interface PostReview {
  rating: number;
  comment: string;
}
