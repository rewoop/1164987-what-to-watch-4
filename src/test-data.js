export const TestFilm = {
  id: 1,
  filmTitle: `The Rock`,
  filmGenre: `Thriller`,
  filmVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  releaseDate: 1996,
  isFavoriteFilm: true,
  runTime: `356`,
  backgroundPoster: `img/bg-the-grand-budapest-hotel.jpg`,
  filmPoster: `img/the-grand-budapest-hotel-poster.jpg`,
  ratingScore: 9,
  ratingLevel: 9,
  ratingCount: `1337 ratings`,
  filmDescription: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  filmDirector: `Wes Andreson`,
  filmStarring: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Tom Cruz`
  ],
};

export const TestFilms = [TestFilm];

export const TestComment = {
  id: 0,
  user: {
    id: 1,
    name: `Kate Muir`,
  },
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `December 24, 2016`,
  rating: 9,
};

export const TestComments = [TestComment];
