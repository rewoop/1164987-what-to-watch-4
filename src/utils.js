import {ALL_GENRES, RatingLevel} from "./const.js";

export const formatReviewDate = (dateString) => {
  const date = new Date(dateString);
  const dateTimeFormat = new Intl.DateTimeFormat(`en`, {year: `numeric`, month: `2-digit`, day: `2-digit`});
  const [{value: month},, {value: day},, {value: year}] = dateTimeFormat.formatToParts(date);
  return `${year}-${month}-${day}`;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getGenresList = (films) => {
  return [ALL_GENRES].concat(Array.from(new Set(films.map((film) => film.filmGenre))));
};

export const formatRatingScoreToLevel = (score) => {
  if (score >= RatingLevel.MIN_BAD && score <= RatingLevel.MAX_BAD) {
    return `Bad`;
  } else if (score > RatingLevel.MAX_BAD && score <= RatingLevel.MAX_NORMAL) {
    return `Normal`;
  } else if (score > RatingLevel.MAX_NORMAL && score <= RatingLevel.MAX_GOOD) {
    return `Good`;
  } else if (score > RatingLevel.MAX_GOOD && score < RatingLevel.MAX_VERY_GOOD) {
    return `Very good`;
  } else {
    return `Awesome`;
  }
};
