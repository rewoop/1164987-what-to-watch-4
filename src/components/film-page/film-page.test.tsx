import * as React from "react";
import * as renderer from "react-test-renderer";
import FilmPage from "./film-page";
import FilmPageOverview from "../film-page-overview/film-page-overview";
import {Router} from "react-router-dom";
import history from "../../history";
import {noop} from "../../utils";
import {Film, Films} from "../../types";
import {TestFilm, TestFilms} from "../../test-data";

const film: Film = TestFilm;
const films: Films = TestFilms;

const renderActiveTab = () => {
  return <FilmPageOverview
    ratingScore={film.ratingScore}
    ratingLevel={film.ratingLevel}
    ratingCount={film.ratingCount}
    filmDescription={film.filmDescription}
    filmDirector={film.filmDirector}
    filmStarring={film.filmStarring}
  />;
};

it(`Should FilmPage render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <FilmPage
            film={film}
            sortedFilms={films}
            activeTab={`overview`}
            renderActiveTab={renderActiveTab}
            setActiveTab={noop}
            isSignIn={`NO_AUTH`}
            onMyListClickHandler={noop}
            isFavoriteStatus={true}
          />
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
