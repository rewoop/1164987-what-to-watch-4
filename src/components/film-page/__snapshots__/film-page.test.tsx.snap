// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmPage render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Rock"
          src="img/bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/login"
            onClick={[Function]}
            style={
              Object {
                "color": "inherit",
                "cursor": "pointer",
                "textDecoration": "none",
              }
            }
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Rock
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thriller
            </span>
            <span
              className="movie-card__year"
            >
              1996
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/film/1/player"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/film/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Rock"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              9
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                1337 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Wes Andreson
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
